wire/message.go

```diff
@@ -7,69 +7,104 @@ package wire
 import (
 	"bytes"
 	"fmt"
-	"io"
-	"unicode/utf8"
-
 	"github.com/pkg/errors"
+	"io"
 
 	"github.com/kaspanet/kaspad/util/daghash"
 )
 
 // MessageHeaderSize is the number of bytes in a kaspa message header.
-// Kaspa network (magic) 4 bytes + command 12 bytes + payload length 4 bytes +
+// Kaspa network (magic) 4 bytes + command 4 byte + payload length 4 bytes +
 // checksum 4 bytes.
-const MessageHeaderSize = 24
-
-// CommandSize is the fixed size of all commands in the common kaspa message
-// header. Shorter commands must be zero padded.
-const CommandSize = 12
+const MessageHeaderSize = 16
 
 // MaxMessagePayload is the maximum bytes a message can be regardless of other
 // individual limits imposed by messages themselves.
 const MaxMessagePayload = (1024 * 1024 * 32) // 32MB
 
+// MessageCommand is a number in the header of a message that represents its type.
+type MessageCommand uint32
+
+func (cmd MessageCommand) String() string {
+	cmdString, ok := messageCommandToString[cmd]
+	if !ok {
+		cmdString = "unknown command"
+	}
+	return fmt.Sprintf("%s [code %d]", cmdString, uint8(cmd))
+}
+
 // Commands used in kaspa message headers which describe the type of message.
 const (
-	CmdVersion         = "version"
-	CmdVerAck          = "verack"
-	CmdGetAddresses    = "getaddr"
-	CmdAddress         = "addr"
-	CmdGetBlockInvs    = "getblockinvs"
-	CmdInv             = "inv"
-	CmdGetData         = "getdata"
-	CmdNotFound        = "notfound"
-	CmdBlock           = "block"
-	CmdTx              = "tx"
-	CmdPing            = "ping"
-	CmdPong            = "pong"
-	CmdFilterAdd       = "filteradd"
-	CmdFilterClear     = "filterclear"
-	CmdFilterLoad      = "filterload"
-	CmdMerkleBlock     = "merkleblock"
-	CmdReject          = "reject"
-	CmdFeeFilter       = "feefilter"
-	CmdGetBlockLocator = "getlocator"
-	CmdBlockLocator    = "locator"
-	CmdSelectedTip     = "selectedtip"
-	CmdGetSelectedTip  = "getseltip"
-	CmdInvRelayBlock   = "invrelblk"
-	CmdGetRelayBlocks  = "getrelblks"
+	CmdVersion MessageCommand = iota
```

Comment:
Better don't use iota here, hardcode the numbers.
iota is just way too easy to by mistake move all the values by one.